src/psyclone/tests/psyir/frontend/fparser2_main_program_handler_test.py

```diff
@@ -94,3 +94,19 @@ def test_main_program_handler(parser, code, expected):
     writer = FortranWriter()
     result = writer(psyir)
     assert expected == result
+
+
+def test_main_program_handler_codeblock(fortran_reader):
+    '''Test the main_program_handler results in a CodeBlock if the input
+    code contains a child Subroutine.'''
+    code = '''Program TestProgram
+
+    contains
+    Subroutine TestSubroutine()
+    End Subroutine
+    End Program'''
+
+    psyir = fortran_reader.psyir_from_source(code)
+    cblock = psyir.children[0]
+    assert isinstance(cblock, CodeBlock)
+    assert "SUBROUTINE" in str(cblock.get_ast_nodes[0])
```

Comment:
I would have expect the whole Program to be in the CodeBlock? Does re-generating the code (use the `fortran_writer` fixture) give the correct output?

Reply:
Yeah - I should probably check more here, my main thing was to check the subroutine wasn't lost but maybe just checking the output from fortran_writer would also be a good idea.